### ISSUE TITLE
vNext finding incorrect Nuget Transitive Dependencies

### ISSUE THREAD

**Issue by james-asebp** [tinglesoftware/dependabot-azure-devops#1247]
Running dependabot extension in DevOps with the new `useUpdateScriptvNext` flag set. We are seeing in the pipeline logs that it is resolving dependencies with versions too low.

For example we use `Microsoft.ApplicationInsights.AspNetCore` `v2.22.0` in a `net6.0` targeted project. One of the **_deep_** down Transitive dependencies under .net 6.0 for this package is `System.Drawing.Common` >= `6.0.0`. 

During the dependency analysis phase the log outputs this as part of the massive json blob

``` json
{
  "Name": "System.Drawing.Common",
  "Version": "4.7.0",
  "Type": "Unknown",
  "EvaluationResult": null,
  "TargetFrameworks": [
	"net6.0"
  ],
  "IsDevDependency": false,
  "IsDirect": false,
  "IsTransitive": true,
  "IsOverride": false,
  "IsUpdate": false,
  "InfoUrl": null
},
```
As you can see the dependency version found is `4.7.0`, which is a `Vulnerable` verion of this package. Thus the remaining part of the process takes over and happily adds the v`4.7.2` version of the package `System.Drawing.Common` to the csproj file as a packageReference. The problem arises because this is a incompatible version to the version required by the deep dependency of `Microsoft.ApplicationInsights.AspNetCore` and subsequently fails our PR build.

**Categorization**
- [X] This is not a permissions issue (Seek help at https://github.com/tinglesoftware/dependabot-azure-devops/discussions/1245)

**Expected behavior**
The expectation that the transitive dependency checking is correctly "walking" the dependency tree to resolve the correct package versions.

**Screenshots**
![image](https://github.com/user-attachments/assets/cae084ba-acef-4b7a-881a-2a1ca56cd2fa)
![image](https://github.com/user-attachments/assets/dc67df26-0d8c-4c70-acec-d98bb2973021)
![image](https://github.com/user-attachments/assets/1abccf27-071f-40c6-a33d-5a8e58351d54)
![image](https://github.com/user-attachments/assets/1181326c-83ac-4df2-adaa-f4641efb534d)
![image](https://github.com/user-attachments/assets/dfc99e78-eca0-4f43-8de1-e3f456a1bb04)

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Version 1.30.2.800

**Additional context**
We are trying out the vNext script to work around the recent nuget auth issues introduced with the removal of the 1.24 docker image

**Comment by rhyskoedijk** [comment]
This might be a issue with dependabot-core itself:
 - [#9693 NuGet updater should try to fix `NU1605` package downgrade errors](https://github.com/dependabot/dependabot-core/issues/9693)

The post-1.24 NuGet auth issue has been backported to the original update script too since 1.30.2, so you _might_ be able to try again with `useUpdateScriptvNext: false` and see if you get a different result. 

If you don't get this issue with vNext disabled, I can dig deeper in to this and see if it is a specific issue with the vNext script. 

If you do get this issue with vNext disabled, then I'm not sure if there is anything practical that can be done in the scope of this project to fix it (assuming it is an issue with `NuGetUpdater.Core` native tool). I would definitely go give that issue an upvote or share your scenario there so it gets more attention to the team.

**Comment by james-asebp** [comment]
I ran our pipeline again with `useUpdateScriptvNext` set to false this morning.
The dependency `System.Drawing.Common` does appear in the giant json blob again, twice, once as `v6.0.0` and again as `v4.7.0`. But later in the log it does not appear in the dependency summary at all and therefore the update process does not attempt to update this dependency.  

This makes me think either the `grouping` logic or the transient dependency vulnerability checking in vNext is triggering the issue. I understand that vNext might be using more of the dependabot-core code but I am having difficulty following the process in source because my ruby experience is null.

**Comment by rhyskoedijk** [comment]
> This makes me think either the `grouping` logic or the transient dependency vulnerability checking in vNext is triggering the issue.
I will have a look in to this and see what I can find.

**Comment by rhyskoedijk** [comment]
I have not been able to reproduce this but have noticed a few new fixes to dependabot-core around transitive dependencies and ensuring dependencies are valid for target framework:

- https://github.com/dependabot/dependabot-core/pull/10343
- https://github.com/dependabot/dependabot-core/pull/10449

This issue _might_ be resolved when 0.272.0 is released and merged here.

**Comment by james-asebp** [comment]
Closing this issue as V2 seems to be working and instead it is logic in the base dependabot code that is instead blocking us.

Opened [#10791](https://github.com/dependabot/dependabot-core/issues/10791) over there.
